src/app/app-routing.module.ts

```diff
@@ -11,7 +15,15 @@ import {TestcasesComponent} from './testcases/testcases.component';
 const routes: Routes = [
   {path: '', component: LoginComponent},
   {path: 'problems', component: ProblemListComponent},
-  {path: 'exams', component: ExamListComponent},
+  { path: 'exams', component: ExamListComponent, },
+  {
+    path: 'exam/:examId', component: ExamIndexComponent,
```

Comment:
should be `path: 'exams/:examId'`
and I think `ExamHomeComponent` might be better.